frontend/src/app/core/services/user.service.ts

```diff
@@ -3,89 +3,133 @@ import { IImagePath } from '@core/models/IImagePath';
 import { INewUser } from '@core/models/INewUser';
 import { IUpdateUser } from '@core/models/IUpdateUser';
 import { ILocalUser, IUser } from '@core/models/IUser';
-import { map } from 'rxjs';
+import { BehaviorSubject, first, Observable, tap } from 'rxjs';
 
 import { HttpInternalService } from './http-internal.service';
+import { NotificationService } from './notification.service';
 
 @Injectable({
     providedIn: 'root',
 })
 export class UserService {
     public routePrefix = '/user';
 
+    private onUserChanged = new BehaviorSubject<IUser | undefined>(undefined);
```

Comment:
I thought we might need to render some component, which uses user, even if user is removed from local storage (undefined). It's not the case, apparently, because the only way for user to be undefined - after logout.